src/hexo/themes/documentation/source/js/scanner-submit.js

```diff
@@ -0,0 +1,474 @@
+        var updates = data.updates;
+        var time = data.time;
+        var version = data.version;
+
+        var totalErrors = 0;
+        var totalWarnings = 0;
+
+        updates.forEach(function (category) {
+            // Still needs to update the UI even if `category.results` is equal to null.
+            if (categoryPass(category)) {
+                updateAsPass(category);
+
+                return;
+            }
+
+            if (!category.results) {
+                // Note the function shouldn't return when category.results length is 0.
+                // It's possible to have new `passed` rules, which are not included in `results`.
+                return;
+            }
+
+            updateErrorItems(category);
+            updateOverallData(category);
+
+            totalErrors += category.statistics.errors;
+            totalWarnings += category.statistics.warnings;
+
+            if (categoryScanComplete(category)) {
+                removeLoader(category);
+            }
+        });
+
+        var codeBlocks = document.querySelectorAll('code');
+
+        for (var i = 0; i < codeBlocks.length; i++) {
+            hljs.highlightBlock(codeBlocks[i]);
+        }
+
+        document.querySelector('#total-errors').innerHTML = totalErrors;
+        document.querySelector('#total-warnings').innerHTML = totalWarnings;
+        document.querySelector('.scan-overview--time .scan-overview__body--purple').innerHTML = time;
+        document.querySelector('.scan-overview--version .scan-overview__body--purple').innerHTML = version;
+    };
+
+    var updateScanFailUI = function (data) {
+        var scanErrorMessageHTML = '<div class="scan-error">\
+        <p>\
+        There was an error and we were only able to partially complete the scan. View the results below or\
+        <a href="https://sonarwhal.com/scanner">perform another scan</a>.\
+    <p>\
+    </div>';
+
+        document.querySelector('#results-container').insertAdjacentHTML('beforebegin', scanErrorMessageHTML);
+        data.categories.forEach(updateFailedItems);
+    };
+
+    var showQueueMessage = function () {
+        queueBlock.style.display = 'block';
+        resultBlock.style.display = 'none';
+    };
+
+    var hideQueueMessage = function () {
+        queueBlock.style.display = 'none';
+        resultBlock.style.display = 'block';
+    };
+
+    var queryAndUpdate = function () {
+        var callback = function (err, response) {
+            var isFinish = response.status === jobStatus.finished;
+            var isError = response.status === jobStatus.error;
+            var isPending = response.status === jobStatus.pending;
+            var isStarted = response.status === jobStatus.started;
+
+            if (err) {
+                console.error(err);
+
+                return;
+            }
+
+            if (isPending) {
+                showQueueMessage();
```

Comment:
The status just can be one, so you should `return` after show the queue message.

Reply:
@sarvaje  It still needs to get to `setTimeout(queryAndUpdate, 1000);`. So shouldn't return yet?

Reply:
You are right, forget this comment :)